bigbite/themer
# [104] Adds node and npm versions to package.json and .node-version file

## Description
Issue https://github.com/bigbite/themer/issues/104.

Adds node and npm versions to the engines property in `package.json` so developers know which versions of node are supported.  Also adds a `.node-version` file so package managers automatically use the recommended version.

I set the recommended node version to 18 as it still has a year left in [it's lifecycle](https://endoflife.date/nodejs).  I didn't go with node version 20 as some of our installed dependencies do not support it in their current versions.  See screenshot:
![image](https://github.com/user-attachments/assets/8f6a1d78-2f02-4b42-aee4-127ff92803c6)


## Change Log



- sets minimum supported node version to 16 and minimum npm version to 8
- adds a `.node-version` file to set the default node version to 18.

## Steps to test



1. cd out of the root directory then cd back into it.
2. If you are using `fnm` as your package manager, the node version should automatically switch to version 18.
3. Delete the `node_modules` folder.
4. Run `npm i`.
5. The packages should install without issues
6. Run `npm watch:dev` and check that the plugin still works.

## Checklist:
